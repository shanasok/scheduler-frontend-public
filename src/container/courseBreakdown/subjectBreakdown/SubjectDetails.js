import React from 'react';
import SubjectName from './SubjectName';
import SubjectSummary from './SubjectSummary';
import LessonRibbon from './LessonRibbon';

class SubjectDetails extends React.Component {

    render() {
        return <div className="subject">
            <SubjectName name={this.props.subject.name} />
            <SubjectSummary summary={this.props.subject.summary} />
            <LessonRibbon lessons={this.props.subject.lessons} />
        </div>
    }
}

export default SubjectDetails;