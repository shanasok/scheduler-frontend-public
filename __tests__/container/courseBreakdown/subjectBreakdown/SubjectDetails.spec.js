import React from 'react';
import SubjectDetails from '../../../../src/container/courseBreakdown/subjectBreakdown/SubjectDetails';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
    const subject = {
        name: "Threading the Machine",
        summary: "testSummary",
        subjectNumber: 1,
    }
    const tree = renderer
        .create(<SubjectDetails subject={subject}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});