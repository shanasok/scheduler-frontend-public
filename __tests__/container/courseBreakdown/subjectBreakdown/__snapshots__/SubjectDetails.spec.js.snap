// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="subject"
>
  <div
    className="subjectName"
  >
    Threading the Machine
  </div>
  <div
    className="subjectSummary"
  >
    testSummary
  </div>
  <div
    className="lessonRibbon"
  />
</div>
`;
